framework/src/media/FileInputDataSource.cpp

```diff
@@ -73,19 +73,23 @@ bool FileInputDataSource::open()
 		case AUDIO_TYPE_MP3:
 		case AUDIO_TYPE_AAC:
 			if (!utils::header_parsing(mFp, audioType, &channel, &sampleRate, NULL)) {
-				medvdbg("header parsing failed\n");
+				meddbg("header parsing failed\n");
+				return false;
+			}
+			if (fseek(mFp, 0, SEEK_SET) != 0) {
+				meddbg("file seek failed error\n");
 				return false;
 			}
 			setSampleRate(sampleRate);
 			setChannels(channel);
 			break;
 		case AUDIO_TYPE_WAVE:
 			if (!utils::header_parsing(mFp, audioType, &channel, &sampleRate, &pcmFormat)) {
-				medvdbg("header parsing failed\n");
+				meddbg("header parsing failed\n");
 				return false;
 			}
-			if (fseek(mFp, WAVE_HEADER_LENGTH, SEEK_SET) != 0) {
-				medvdbg("file seek failed error\n");
+			if (fseek(mFp, 0, SEEK_SET) != 0) {
```

Comment:
is that right?
That function will do it already(header_parsing).
	if (fseek(fp, 0, SEEK_SET) != 0) {
		meddbg("file seek failed error\n");
		return false;
	}

Reply:
We do not need to (fseek(mFp, 0, SEEK_SET) != 0) again, as you said it's done in header_parsing(), but below snippet should be removed, I think. otherwise, audio_decoder.cpp can not parse wave header. 
```cpp
if (fseek(mFp, WAVE_HEADER_LENGTH, SEEK_SET) != 0) {
	medvdbg("file seek failed error\n");
	return false;
}
```